src/treeherder/job_handler.js

```diff
@@ -71,26 +71,59 @@ const EVENT_MAP = {
   [events.taskException().exchange]: 'exception'
 };
 
+/**
+ * Parses a task route and inspects task.extra.treeherder for revision information.
+ * Attempt to use task.extra.treeherder.revision if it exists.  If not,
+ * use whatever hash is found in the routing key.
+ * Note: This is meant as a transitional period between
+ * using task.extra and relying soley on the routing key.  Routing
+ * key in the future will be the preferred and only method.
+ */
 export function parseTaskRevisionHash(task, prefix) {
-  // If task.extra.revision and task.extra.revision_hash exist, use those
-  // for submitting treeherder jobs instead of parsing revision hash from
-  // the route
-  if (task.extra && task.extra.treeherder) {
-    let treeherder = task.extra.treeherder
-    if (treeherder.revision && treeherder.revision_hash) {
-      return [treeherder.revision, treeherder.revision_hash];
-    }
+  let revision = _.get(task, 'extra.treeherder.revision');
+  let revision_hash = _.get(task, 'extra.treeherder.revision_hash');
+
+  if (revision && revision_hash) {
+    return [revision, revision_hash];
   }
 
-  // Find treeherder specific route
-  let route = task.routes.find((route) => {
+  // Find treeherder specific routes
+  let routes = task.routes.filter((route) => {
```

Comment:
so task.routes.filter takes a function, I'm passing in an anonymous function
```js
(route) => {
  return route.split('.')[0] === prefix;
}
```

Basically it iterates over an array, and passes that element to the function you provide, the function then returns true or false.  All elements that evaluated to true will be added to an array and returned, which is assigned to "routes" in this case.

What this is checking for are any routes that begin with the prefix mozilla-taskcluster is listening for, such as "tc-treeherder".  The reason for the change between "find" and "filter" in this PR is that during a transition a task might have both routes in it whereas "find" will only return the first element that evaluates true.

Thinking about this more though, maybe we wouldn't have both routes in a single task.  Maybe it's either the v1 route or the v2, but not both.  I suppose it doesn't hurt handling the case where both are in a single task though.